validator/validation_entry.go

```diff
@@ -21,4 +25,15 @@ type ValidationInput struct {
 	DelayedMsg    []byte
 	StartState    GoGlobalState
 	DebugChain    bool
+
+	SelfHash string // Is a unique identifier which can be used to compare any two instances of validationInput
+}
+
+// SetSelfHash should be only called once. In the context of redis streams- by the producer, before submitting a request
+func (v *ValidationInput) SetSelfHash() {
```

Comment:
if selfhash is empty we don't need to exit early instead just carry on as usual without having the unique identifier, this does imply that workers might work on the same request from multiple producers but there's no reason to stop accepting request altogether just because they couldn't be uniquely identified

Reply:
sorry, I meant exiting early if `SelfHash` is not empty, so we would have:
```
func (v *ValidationInput) SetSelfHash() {
    if v.SelfHash != "" {
        log.Error("SetSelfHash called more then once") // we may even want log warn or error here, as it shouldn't happen unless we have some bug in the code using SetSelfHash
        return // exiting early as hash has already been set
    }
```
Does it make sense?

Reply:
Yep, it makes sense, will include this change